src/test/java/com/unstoppabledomains/resolution/ResolutionTest.java

```diff
@@ -201,4 +203,52 @@ public void dnsRecordsToMap() throws Exception {
     }
+
+    @Test
+    public void passingCustomProvider() throws Exception {
+        IOException cause = new IOException("for testing purposes");
+        IProvider provider = new IProvider() {
+
+            @Override
+            public JsonObject post(String url, JsonObject body) throws IOException {
+                throw cause;
+            }
+            
+        };
+        Resolution resolutionWithProvider = Resolution.builder().provider(provider).build();
+        TestUtils.checkError(
+            () -> resolutionWithProvider.getAddress("brad.crypto", "eth"),
+            NSExceptionCode.BlockchainIsDown,
+            cause
+        );
+    }
+
+    @Test
+    public void passingCorrectProvider() throws Exception {
+        IProvider provider = new IProvider() {
+            @Override
+            public JsonObject post(String url, JsonObject body) throws IOException {
+                if (body.has("params")) {
+                    JsonArray params = body.getAsJsonArray("params");
+                    JsonObject object = params.get(0).getAsJsonObject();
+                    if (
+                        object.has("data") &&
+                        object.get("data").getAsString()
+                            .equals("0x91015f6b0000000000000000000000000000000000000000000000000000000000000040756e4e998dbffd803c21d23b06cd855cdc7a4b57706c95964a37e24b47c10fc900000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000001263727970746f2e4554482e616464726573730000000000000000000000000000")
+                    ) {
+                        JsonObject answer = new JsonObject();                        
+                        answer.addProperty("jsonrpc", "2.0");
+                        answer.addProperty("id", "1");
+                        answer.addProperty("method", "eth_call");
+                        answer.addProperty("result", "0x000000000000000000000000b66dce2da6afaaa98f2013446dbcb0f4b0ab28420000000000000000000000008aad44321a86b170879d7a244c1e8d360c99dda8000000000000000000000000000000000000000000000000000000000000006000000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000002a30783861614434343332314138366231373038373964374132343463316538643336306339394464413800000000000000000000000000000000000000000000");
+                        return answer;
+                    }
+                }
+                throw new IOException("body has incorrect data in the test");
+            }
+        };
+        Resolution resolutionWithProvider = Resolution.builder().provider(provider).build();
+        String ethAddress = resolutionWithProvider.getAddress("brad.crypto", "eth");
+        assertEquals("0x8aaD44321A86b170879d7A244c1e8d360c99DdA8", ethAddress);
+    }
```

Comment:
@JohnnyJumper Please, add test for default provider. You can use https://httpbin.org for testing.